presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -212,11 +212,11 @@ public Expression rewriteExpression(Expression node, NodeContext context, Expres
         @Override
         public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, NodeContext context, ExpressionTreeRewriter<NodeContext> treeRewriter)
         {
-            List<Expression> predicates = extractPredicates(node.getType(), node).stream()
-                    .map(expression -> treeRewriter.rewrite(expression, NodeContext.NOT_ROOT_NODE))
-                    .collect(toImmutableList());
+            Expression expression = combinePredicates(node.getType(), extractPredicates(node.getType(), node).stream()
```

Comment:
Yea, I think this does the same thing. I'm impartial to either approach. If we go with this suggestion, we should leave a comment as to why we need to re-extract the predicates from the result.